assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js

```diff
@@ -84,6 +84,7 @@ export type PaymentAuthorisation = StripeAuthorisation | PayPalAuthorisation | D
 // because the end of that checkout happens on the backend after the user is redirected to our site.
 export type PaymentResult
   = {| paymentStatus: 'success' |}
+  | {| paymentStatus: 'success', isPending: true |}
```

Comment:
Any reason you prefer this to: `{| paymentStatus: 'pending' |}`

I find it slightly misleading to mark the paymentStatus as success unless we know for sure that it has succeeded!

Reply:
not really, shall we go with : pending?

Reply:
Now that we are using this to handle subscriptions as well as contributions, the following might be more appropriate:

```
export type CheckoutResult
= {| checkoutStatus: 'success' |}	
| {| checkoutStatus: 'success', subscriptionCreationPending: true |}
| {| checkoutStatus: 'failure', error: ErrorReason |};
```

Unlike contributions, the first payment for subscriptions products is not always taken immediately anyway (e.g. Digital Pack free trial).